### ISSUE TITLE
Scopes from request object get compared to reduced list of scopes during authorization

### ISSUE THREAD

**Issue by LorisBachert** [GluuFederation/oxAuth#759]
When passing a request object during authorization the scopes i passed as query parameter and the scopes i passed inside my request object get compared. Unfortunately the scopes i passed as query parameter get filtered by checking the policy. Therefore the scopes differ and i get an exception.
So, for example, if i pass the scopes "profile email" as url paramter and in my request object i have `scopes: [ "profile", "email" ]`. Since i'm not allowed to access the email my scopes from the url get reduced to "profile" and afterwards they are compared to the non-reduced scopes from the request object.
Here is the code:
```java
List<String> scopes = new ArrayList<String>();
if (StringHelper.isNotEmpty(scope)) {
	Set<String> grantedScopes = scopeChecker.checkScopesPolicy(client, scope);
	scopes.addAll(grantedScopes);
}
```
and later:
```java
if (!jwtAuthorizationRequest.getScopes().containsAll(scopes)
		|| !scopes.containsAll(jwtAuthorizationRequest.getScopes())) {
	throw new InvalidJwtException("The scope parameter is not the same in the JWT");
}
```

**Comment by nynymike** [comment]
Not sure why you would use both query params and request object. Should this even be allowed?

**Comment by yuriyz** [comment]
According to spec request object scopes always win (same as other parameters). There is condition that query scope must always contain `openid` scope to indicate that it is Connect request. It's already fixed in 4.2 as part of FAPI fixes.
